UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/NMDataService.cs

```diff
@@ -226,31 +226,58 @@ public async Task<List<ShowDailyFilesResponseModel>> GetDailyBatchDetailsFiles(s
             }
         }
 
-        public async Task<List<ShowFilesResponseModel>> GetLeisureFilesAsync(string correlationId)
+        public async Task<ShowNMFilesResponseModel> GetLeisureFilesAsync(string correlationId)
         {
             try
             {
-                string accessToken = await _authFssTokenProvider.GenerateADAccessToken(_userService.IsDistributorUser, correlationId);
+                BatchSearchResponse searchResult = new();
+                bool isCached = false;
+                const string frequency = "leisure";
+                const string partitionKey = "Public";
+                const string rowKey = "LeisureKey";
+                if (_cacheConfiguration.Value.IsFssCacheEnabled)
+                {
+                    BatchSearchResponseModel batchSearchResponseModel = await _fileShareServiceCache.GetBatchResponseFromCache(partitionKey, rowKey, frequency, correlationId);
 
-                _logger.LogInformation(EventIds.ShowLeisureFilesResponseStarted.ToEventId(), "Request to get leisure files started with _X-Correlation-ID:{correlationId}", correlationId);
+                    if (batchSearchResponseModel.BatchSearchResponse != null)
+                    {
+                        searchResult = batchSearchResponseModel.BatchSearchResponse;
+                        isCached = true;
+                    }
+                }
+                if (searchResult.Entries == null)
+                {
+                    string accessToken = await _authFssTokenProvider.GenerateADAccessToken(_userService.IsDistributorUser, correlationId);
 
-                const string searchText = $" and $batch(Frequency) eq 'leisure'";
+                    _logger.LogInformation(EventIds.ShowLeisureFilesResponseStarted.ToEventId(), "Request to get leisure files started with caching data:{isCached} and _X-Correlation-ID:{correlationId}", isCached, correlationId);
 
-                IFileShareApiClient fileShareApiClient = new FileShareApiClient(_httpClientFactory, _fileShareServiceConfig.Value.BaseUrl, accessToken);
+                    const string searchText = $" and $batch(Frequency) eq 'leisure'";
```

Comment:
Just one minor change, use constant instead of hardcoded string.
```suggestion
                    const string searchText = $" and $batch(Frequency) eq '{frequency}'";
```

Reply:
resolved